workflows/phaseimpute/main.nf

```diff
@@ -396,6 +406,46 @@ workflow PHASEIMPUTE {
             ch_input_validate = ch_input_validate.mix(CONCAT_QUILT.out.vcf_tbi)
         }
 
+        if (params.tools.split(',').contains("beagle5")) {
+            log.info("Impute with BEAGLE5")
+
+            // Prepare VCF input channel with clean metadata
+            ch_vcf_clean = ch_input_type.vcf
+                .map { meta, vcf, index -> 
+                    [[id: meta.id], vcf, index]
+                }
+
+            // Extract chromosome information from VCF
+            VCFCHREXTRACT_BEAGLE5(ch_vcf_clean.map { meta, vcf, index -> [meta, vcf] })
+            
+            // Create per-chromosome channels for BEAGLE5
+            ch_beagle5_input = VCFCHREXTRACT_BEAGLE5.out.chr
+                .join(ch_vcf_clean, by: 0)
+                .map { meta, chr_file, vcf, index ->
+                    chr_file.readLines().collect { chr ->
+                        [[id: meta.id, chr: chr], vcf, index]
+                    }
+                }
+                .flatten()
+                .collate(3)
```

Comment:
Same as in the other PR.
We need to discuss how do we parallelize across the chromosomes